### ISSUE TITLE
FR: Audio Loudness Normalization OR Replay Gain Support

### ISSUE THREAD

**Issue by X-Raym** [UltraStar-Deluxe/USDX#352]
Hi,

### Actual behaviour

Some mp3 can have very different volume level. This means that in a game, you may have to keep your hand on the speaker volume and adjust at every new song.

### Expected behaviour

It would be nice if there was a Loudness Normalization in UltraStar, so that the Loudness will be auto-adjusted at song start and so we don't have to adjust volume manually.

It can be handle by processing the song in UltraStar directly, or by bringing support to mp3 tag data like [https://www.lifewire.com/how-to-normalize-mp3-files-2438861](Replay Gain) and third party software to take care of this metadata generation (like [MP3](https://www.lifewire.com/how-to-normalize-mp3-files-2438861gain] )

Thanks !

**Comment by bohning** [comment]
I fully agree, this is an issue I have been 'struggling' with for years now and have become used to regulating the volume up and down constantly. It would be nice to have USDX handle the loudness of the song or adjust to existing normalization tags.

**Comment by GrueneNeun** [comment]
I actually use MP3Gain -> http://mp3gain.sourceforge.net/ on new Songs before adding them to my library and didnt have to touch the volume in ages.

**Comment by bohning** [comment]
@GrueneNeun Yes, unfortunately mp3gain is only available for Windows :-(

**Comment by X-Raym** [comment]
@GrueneNeun is there an option to activate support of this ? I didn't see any diff between before process and after. But maybe I missed something.
Is it related to the Music Gain option ?

**Comment by GrueneNeun** [comment]
Not that i knew of.
I use it to normalize all MP3s to 89,0dB. Just to make sure - you did use the "Track Gain" Button and not just the "Track Analysis" one ?

**Comment by X-Raym** [comment]
@GrueneNeun I can't run MPgain (bug because of missing mscomctl.ocx), I use wxMP3gain [instead](https://sourceforge.net/projects/wxmp3gain/). But I didn't see any chance before/after, I suspect there is a bug during the metadata writting process. I tried QMP3 instead, but ti shows me what's wrong: with the default -89db value, no ReplayGain is set. But with -90, some tracks get a +1.5db Boost. There is some weird rounding here.

[Demo](https://i.imgur.com/xqLClaU.gif)

It doesn't seem that these ReplayGain are precized at the decimal level.

Anyway, I test with a mp3 and a copu of this mp3 file 10db lower, process with ReplayGain, and I didn't see any change.

The Music Gain option seems to be a fixed dB boost no matter the source.

**Comment by GrueneNeun** [comment]
Sorry, but i dont know how these frontends work with MP3Gain - do they just modify the tags oder do they really apply the volume to the files ?

**Comment by X-Raym** [comment]
@GrueneNeun File volume doesn't change (if you check waveform), so I suppose it is just a tag which is changed.
Thanks for your help anyway :P
